src/v/coproc/script_context.cc

```diff
@@ -177,34 +178,30 @@ ss::future<std::vector<process_batch_request::data>> script_context::read() {
       });
 }
 
-std::optional<storage::log_reader_config>
+storage::log_reader_config
 script_context::get_reader(const ss::lw_shared_ptr<ntp_context>& ntp_ctx) {
     auto found = ntp_ctx->offsets.find(_id);
     vassert(
       found != ntp_ctx->offsets.end(),
       "script_id must exist: {} for ntp: {}",
       _id,
       ntp_ctx->ntp());
-    const auto& cp_offsets = found->second;
-    if (cp_offsets.last_read != cp_offsets.last_acked) {
+    const ntp_context::offset_pair& cp_offsets = found->second;
+    const model::offset next_read
+      = (unlikely(cp_offsets.last_acked == model::offset{}))
+          ? model::offset(0)
```

Comment:
why not initialize the `last_acked=0` ?

Reply:
If you look up this is the exact opposite to Michals suggestion. His point is that if you never last_read or last_acked the 0 offset, you shouldn't be initializing it to 0. Thoughts?

Reply:
the way i think about this stuff is that the default value of model::offset can be treated as if you were instead passing around a std::optional<model::offset>. it the natural initial value for the optional is nullopt, then the natural initial value for the raw model::offset should be its built-in default value.